poc/extensions/src/perm_controller.rs

```diff
@@ -0,0 +1,18 @@
+use crate::ExtensionIdTy;
+pub trait PermController {
+    fn is_allowed(extension_id: ExtensionIdTy, call: Vec<u8>, source: &InvokeSource) -> bool;
```

Comment:
```suggestion
    fn is_allowed(extension_id: ExtensionIdTy, call: &[u8], source: &InvokeSource) -> bool;
```

pass reference if possible